src/config.rs

```diff
@@ -184,28 +188,84 @@ impl ApplicationConfigBuilder {
             diagnostic_mode: false,
             log_file_size: 10 * 1024,
             log_file_rotations: 1,
+            pid_file: None,
+            lock_file: None,
         };
 
         Ok(builder)
     }
 
-    /// Build application configuration.
-    fn build(self) -> Result<ApplicationConfig, ConfigError> {
-        let mut logger = match self.logger_type {
+    /// Create a lock file (if specified).
+    fn create_lock_file(&self) -> Result<Option<File>, ConfigError> {
+        self.lock_file
+            .as_ref()
+            .map(|lock_file| {
+                File::create(lock_file)
+                    .and_then(|mut lock_file| {
+                        lock_file.try_lock_exclusive()?;
+                        lock_file.write_fmt(format_args!("{}\n", process::id()))?;
```

Comment:
Does it make sense to have two options which both store PID into the file? In my point of view it only makes sense to have `lock-file` option and the `pid-file` can be removed. Or is there a reasonable use-case when `lock-file` can't be used to store the PID?

Reply:
The idea was to give users an option not to create a lock file. But I'm not sure if there will be a use case for it. I'll remove the PID file for now. We can add it any time in the future again if there is a need for it.